client/me/profile-links/data/use-add-profile-link-mutation.ts

```diff
@@ -0,0 +1,54 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { useCallback } from 'react';
+import wp from 'calypso/lib/wp';
+import type {
+	AddProfileLinksMutationOptions,
+	AddProfileLinksPayload,
+	AddProfileLinksResponse,
+	AddProfileLinksVariables,
+	ProfileLink,
+} from './types';
+
+export const useAddProfileLinkMutation = ( options: AddProfileLinksMutationOptions = {} ) => {
+	const queryClient = useQueryClient();
+	const mutation = useMutation( {
+		mutationFn: ( { links }: AddProfileLinksVariables ): Promise< AddProfileLinksResponse > => {
+			return wp.req.post( `/me/settings/profile-links/new`, { apiVersion: '1.2' }, { links } );
+		},
+		...options,
+		onSuccess: ( ...args ) => {
+			const [ data ] = args;
+
+			queryClient.setQueryData(
+				[ 'profile-links' ],
```

Comment:
Are we clearing this cache properly when the user logs out / logs in? Should we key them by user ID?